### PR TITLE
fix(building): allow all states to be set for new buildings

### PR DESCRIPTION
**Pull request by luytena**
Allow buildings to be created with any initial status. Depending
on the selected status, different API requests are emitted.

### CODE REVIEW

File: addon/controllers/building/edit/form.js [hunk 1]
```diff
@@ -79,10 +75,19 @@ export default class BuildingFormController extends Controller {
...
-          this.buildingWork.building
-        );
-        EGID = building.EGID;
+        const buildingStatus = this.buildingWork.building.buildingStatus;
+        if (buildingStatus !== EXISTING && buildingStatus !== NOT_USABLE) {
+          const building = yield this.buildingAPI.createAndAddToConstructionProject(
+            this.model.projectId,
+            this.buildingWork
+          );
+          EGID = building.EGID;
```
Comment (velrest) [inline addon/controllers/building/edit/form.js:84]: This can be moved outside of the `if ... else` to reduce some duplication.

File: addon/models/construction-project.js [hunk 2]
```diff
@@ -155,7 +155,6 @@ export default class ConstructionProject extends XMLModel {
     {{/if}}
 
     {{#if model.isNew}}
-      {{! TODO}}
```
Comment (velrest) [inline addon/models/construction-project.js:158]: Why did you remove the todo? We still have to find a way around this.

Comment (velrest) [reply addon/models/construction-project.js:158]: Or can we remove the empty `<work>` in the new version?

File: addon/services/building.js [hunk 3]
```diff
@@ -95,6 +96,56 @@ export default class BuildingService extends GwrService {
     return this.createAndCache(xml);
   }
 
+  async createAndAddToConstructionProject(EPROID, buildingWork) {
+    let body = this.xml.buildXMLRequest("addWorkToProject", buildingWork);
+    let response = await this.authFetch.fetch(
+      `/constructionprojects/${EPROID}/work`,
```
Comment (velrest) [inline addon/services/building.js:102]: what is the difference between this and the call in `.create`?

File: addon/models/construction-project.js (hunk 2 above)
Comment (luytena) [reply addon/models/construction-project.js:158]: Nope, you're correct, I will return it.

File: addon/services/building.js (hunk 3 above)
Comment (luytena) [reply addon/services/building.js:102]: It allows buildings to be added that are not in the state existing or unusable. This is all the information I found:
image

**Review by velrest — APPROVED** [review]
